src/test/metrics_test.cpp

```diff
@@ -0,0 +1,414 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gtest/gtest.h"
+
+#include "ppx/metrics.h"
+
+#if !defined(NDEBUG)
+#define PERFORM_DEATH_TESTS
+#endif
+
+namespace ppx {
+
+////////////////////////////////////////////////////////////////////////////////
+// Fixture
+////////////////////////////////////////////////////////////////////////////////
+
+class MetricsTestFixture : public ::testing::Test
+{
+protected:
+    void SetUp() override
+    {
+        manager = new metrics::Manager();
+        ASSERT_NE(manager, nullptr);
+        run = manager->AddRun("default_run");
+        ASSERT_NE(run, nullptr);
+    }
+
+    void TearDown() override
+    {
+        if (manager != nullptr) {
+            delete manager;
+            manager = nullptr;
+        }
+    }
+
+protected:
+    metrics::Manager* manager;
+    metrics::Run*     run;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Manager
+////////////////////////////////////////////////////////////////////////////////
+
+TEST(MetricsTest, ManagerAddSingleRun)
+{
+    metrics::Manager manager;
+    metrics::Run*    run = manager.AddRun("run");
+    EXPECT_NE(run, nullptr);
+}
+
+#if defined(PERFORM_DEATH_TESTS)
+TEST(MetricsTest, ManagerAddRunWithNullName)
+{
+    metrics::Manager manager;
+    EXPECT_DEATH({
+        metrics::Run* run = manager.AddRun(nullptr);
+    },
+                 "");
+}
+#endif
+
+TEST(MetricsTest, ManagerAddMultipleRun)
+{
+    metrics::Manager manager;
+
+    metrics::Run* run0;
+    metrics::Run* run1;
+
+    {
+        run0 = manager.AddRun("run0");
+        ASSERT_NE(run0, nullptr);
+    }
+    {
+        run1 = manager.AddRun("run1");
+        ASSERT_NE(run1, nullptr);
+    }
+    ASSERT_NE(run0, run1);
+}
+
+#if defined(PERFORM_DEATH_TESTS)
+TEST(MetricsTest, ManagerAddDuplicateRun)
+{
+    metrics::Manager      manager;
+    constexpr const char* RUN_NAME = "run";
+    {
+        metrics::Run* run = manager.AddRun(RUN_NAME);
+        EXPECT_NE(run, nullptr);
+    }
+    {
+        EXPECT_DEATH({
+            manager.AddRun(RUN_NAME);
+        },
+                     "");
```

Comment:
This is clang format going wild I suppose?

Reply:
Yeah :(